libandroid-navigation/src/main/java/com/mapbox/services/android/navigation/v5/navigation/NavigationService.java

```diff
@@ -311,6 +311,16 @@ void sendRerouteEvent(SessionState sessionState) {
 
     NavigationMetricsWrapper.rerouteEvent(sessionState, routeProgress,
       sessionState.lastRerouteLocation());
+
+    for (SessionState session : queuedRerouteEvents) {
+      session.toBuilder().lastRerouteDate(
+        sessionState.rerouteDate()
```

Comment:
I'm not sure what's happening here. Won't this modify all outstanding reroute events with the date of the reroute event that's about to be sent?

Reply:
I need to update all the sessionState's in the `queuedRerouteEvents` with the latest reroute date.

Reply:
secondsSinceLastReroute should be the seconds from the reroute before the captured reroute event, rather than the most recent reroute event. If a reroute happens while an old reroute event is on the queue, this will overwrite that information.